### PR TITLE
BUPH-104 | per actor template ordering

### PR DESCRIPTION
**Pull request by jpmarcotte**

### CODE REVIEW

File: src/V1/FabricationFile/Actor.php [hunk 1]
```diff
@@ -163,6 +164,31 @@ public function setTemplateRelativeFilePath(string $TemplateRelativeFilePath): A
...
+
+        return $this->PreferredTemplateTrees;
+    }
+
+    public function hasPreferredTemplateTrees(): bool
+    {
+        return $this->PreferredTemplateTrees !== null;
+    }
+
+    public function setPreferredTemplateTrees($PreferredTemplateTrees): ActorInterface
```
Comment (rhift) [inline src/V1/FabricationFile/Actor.php]: missing parameter type =)

File: src/V1/FabricationFile/Actor.php [hunk 2]
```diff
@@ -163,6 +164,31 @@ public function setTemplateRelativeFilePath(string $TemplateRelativeFilePath): A
         return $this;
     }
 
+    public function getPreferredTemplateTrees()
```
Comment (rhift) [inline src/V1/FabricationFile/Actor.php]: missing return type =)

File: src/V1/Actor/Template/Builder.php [hunk 3]
```diff
@@ -57,6 +57,24 @@ private function buildActorTemplateFilePathCandidates(): array
...
+                        sprintf('Template tree %s referenced in %s > %s has not been defined.',
+                            $treeName,
+                            $this->getFabricationFile()->getFilePath(),
+                            $this->getFabricationFileActor()->getFileName()
+                        )
+                    );
+                }
+            }
+        }
+
```
Comment (rhift) [inline src/V1/Actor/Template/Builder.php]: nice =)

File: src/V1/FabricationFile/Actor.php (hunk 2 above)
Comment (rhift) [inline src/V1/FabricationFile/Actor.php]:
```suggestion
    public function getPreferredTemplateTrees(): array
```

File: src/V1/FabricationFile/ActorInterface.php [hunk 4]
```diff
@@ -28,4 +28,10 @@ public function getTemplateRelativeDirectoryPath(): string;
     public function setTemplateFileExtension(string $TemplateFileExtension): ActorInterface;
 
     public function getTemplateFileExtension(): string;
+
+    public function getPreferredTemplateTrees();
```
Comment (rhift) [inline src/V1/FabricationFile/ActorInterface.php]:
```suggestion
    public function getPreferredTemplateTrees(): array;
```

File: src/V1/FabricationFile/Actor.php (hunk 2 above)
Comment (jpmarcotte) [reply src/V1/FabricationFile/Actor.php]: *doh* Thanks. I forgot about the return types.

**Pull request merged**